apps/judicial-system/backend/src/app/modules/indictment-count/indictmentCount.service.ts

```diff
@@ -64,21 +66,28 @@ export class IndictmentCountService {
   }
 
   async delete(caseId: string, indictmentCountId: string): Promise<boolean> {
-    const numberOfAffectedRows = await this.indictmentCountModel.destroy({
-      where: { id: indictmentCountId, caseId },
+    this.sequelize.transaction(async (transaction) => {
+      const numberOfAffectedRows = await this.indictmentCountModel.destroy({
+        where: { id: indictmentCountId, caseId },
+        transaction,
+      })
+      await this.offenseModel.destroy({
```

Comment:
Please note that you need to delete from the offense table first to avoid foreign key exists errors.